Sources/Classes/RSMoEngageDestination.swift

```diff
@@ -6,62 +6,89 @@
 //
 
 import Foundation
+
 import Rudder
-import MoEngage
+import MoEngageSDK
+
+
 
 class RSMoEngageDestination: NSObject, RSDestinationPlugin, UNUserNotificationCenterDelegate {
     let type = PluginType.destination
     let key = "MoEngage"
     var client: RSClient?
     var controller = RSController()
+    var moEngageDataCenter: MoEngageDataCenter?
 
-    func update(serverConfig: RSServerConfig, type: UpdateType) {
+    func update(serverConfig: RSServerConfig, type: UpdateType)  {
         guard type == .initial else { return }
         guard let moEngageConfig: RSMoEngageConfig = serverConfig.getConfig(forPlugin: self) else {
             client?.log(message: "Failed to Initialize MoEngage Factory", logLevel: .warning)
             return
         }
+        
+        // Redirect data according to region
+        if moEngageConfig.region == "EU" {
+            moEngageDataCenter = .data_center_02
+        } else if moEngageConfig.region == "US" {
+            moEngageDataCenter = .data_center_01
+        } else if moEngageConfig.region == "IND" {
+            moEngageDataCenter = .data_center_03
+        }
+        else{
+            client?.log(message:"MoEngage SDK initialization terminated due to an invalid region.", logLevel: .warning)
+        }
+       
+        let sdkConfig = MoEngageSDKConfig(appId: moEngageConfig.apiId, dataCenter: moEngageDataCenter!)
+        
+        //enable debugging
+        if client?.configuration?.logLevel != .none {
+            MoEngageCore.MoEngageConsoleLogConfig(isLoggingEnabled: true, loglevel: MoEngageCore.MoEngageLoggerType.debug)
+        }
```

Comment:
Please follow [this](https://github.com/rudderlabs/rudder-integration-moengage-swift/pull/4/commits/c45dbf7b757f6aa28e3fd95c0184c546855af665) commit to properly implement the MoEngage log level.

As of now, the log level is not even set.

Reply:
@shwetalodhe123 I seem to have made a mistake in my commit (which I've referred above). Please follow this updated logic:
```
if client?.configuration?.logLevel != .none {
    sdkConfig.consoleLogConfig = getMoEngageLogLevel(client?.configuration?.logLevel)
}
```

```
func getMoEngageLogLevel(logLevel: RSLogLevel) -> MoEngageCore.MoEngageConsoleLogConfig {
        if (logLevel == RSLogLevel.error) {
            return MoEngageCore.MoEngageConsoleLogConfig(isLoggingEnabled: true, loglevel: MoEngageCore.MoEngageLoggerType.error)
        } else if (logLevel == RSLogLevel.warning) {
            return MoEngageCore.MoEngageConsoleLogConfig(isLoggingEnabled: true, loglevel: MoEngageCore.MoEngageLoggerType.warning)
        } else if (logLevel == RSLogLevel.info) {
            return MoEngageCore.MoEngageConsoleLogConfig(isLoggingEnabled: true, loglevel: MoEngageCore.MoEngageLoggerType.info)
        } else if (logLevel == RSLogLevel.debug) {
            return MoEngageCore.MoEngageConsoleLogConfig(isLoggingEnabled: true, loglevel: MoEngageCore.MoEngageLoggerType.verbose)
        }
    }
```

Please make sure to test this logic.